### ISSUE TITLE
The confusing error message about the argument while loading a model using scripting

### ISSUE THREAD

**Issue by kevin-steiner** [pytorch/pytorch#22964]
## 🐛 Bug

When I try to scripting my model. it comes up an error message about the arguments.
## To Reproduce
![image](https://user-images.githubusercontent.com/29630934/61360754-a5738580-a8b1-11e9-9855-50f6602c65ae.png)

**The script code:**
```
     @torch.jit.script_method
     def forward(self, w_in, c_in, mask):
         """
         Sequence labeling model.
 
         Parameters
         ----------
         w_in : ``torch.LongTensor``, required.
             The RNN unit.
         c_in : ``torch.LongTensor`` , required.
             The number of characters.
         mask : ``torch.ByteTensor`` , required.
             The mask for character-level input.
         """
         w_emb = self.word_embed(w_in)
 
         c_emb = self.char_embed(c_in)
 
         emb = self.drop( torch.cat([w_emb, c_emb], 2) )
 
         out = self.rnn(emb)
 
         mask = mask.unsqueeze(2).expand_as(out)
 
         out = out.masked_select(mask).view(-1, self.rnn_outdim)
 
         return out
```

## Expected behavior
   Can run the code without error

## Environment

 - PyTorch Version (e.g., 1.0):1.0.0
 - OS (e.g., Linux):Linux 2.2
 - Python version: 3.7
 - CUDA/cuDNN version: 8.0
 - GPU models and configuration: not used
 - Any other relevant information: 

## Additional context
**masked_select**  has the same problem.

**Comment by jamesr66a** [comment]
Hi @kevin-steiner, what type of module is `self.rnn`? The error message is saying `out` is a tuple, and that would make sense if `self.rnn` is something like LSTM

**Comment by kevin-steiner** [comment]
> Hi @kevin-steiner, what type of module is `self.rnn`? The error message is saying `out` is a tuple, and that would make sense if `self.rnn` is something like LSTM

the rnn code is :
```
class BasicUnit(nn.Module):
    """
    The basic recurrent unit for the vanilla stacked RNNs.

    Parameters
    ----------
    unit : ``torch.nn.Module``, required.
        The type of rnn unit.
    input_dim : ``int``, required.
        The input dimension fo the unit.
    hid_dim : ``int``, required.
        The hidden dimension fo the unit.
    droprate : ``float``, required.
        The dropout ratrio.
    batch_norm: ``bool``, required.
        Incorporate batch norm or not. 
    """
    def __init__(self, unit, input_dim, hid_dim, droprate, batch_norm):
        super(BasicUnit, self).__init__()

        self.unit_type = unit
        rnnunit_map = {'rnn': nn.RNN, 'lstm': nn.LSTM, 'gru': nn.GRU}
        self.layer = nn.LSTM(input_dim, hid_dim//2, 1, batch_first=True, bidirectional=True)#need consider
        
        self.droprate = droprate
        self.batch_norm = batch_norm
        if self.batch_norm:
            self.bn = nn.BatchNorm1d(hid_dim)
        self.output_dim = hid_dim
```
```
class BasicRNN(nn.Module):
    """
    The multi-layer recurrent networks for the vanilla stacked RNNs.

    Parameters
    ----------
    layer_num: ``int``, required.
        The number of layers. 
    unit : ``torch.nn.Module``, required.
        The type of rnn unit.
    input_dim : ``int``, required.
        The input dimension fo the unit.
    hid_dim : ``int``, required.
        The hidden dimension fo the unit.
    droprate : ``float``, required.
        The dropout ratrio.
    batch_norm: ``bool``, required.
        Incorporate batch norm or not. 
    """
    def __init__(self, layer_num, unit, emb_dim, hid_dim, droprate, batch_norm):
        super(BasicRNN, self).__init__()

        self.layer_list = [BasicUnit(unit, emb_dim, hid_dim, droprate, batch_norm)] + [BasicUnit(unit, hid_dim, hid_dim, droprate, batch_norm) for i in range(layer_num - 1)]
        self.layer = nn.Sequential(*self.layer_list)
        self.output_dim = self.layer_list[-1].output_dim

        self.init_hidden()
```
 Yes, it is LSTM.

**Comment by kevin-steiner** [comment]
**One more question, If I skip this two error (just comment them), I get an error message with:**
```
RuntimeError: 
could not export python function call <python_value>. Remove calls to python functions before export.:

        def forward(self, input):
            for m in self:
                input = m(input)
                        ~ <--- HERE
            return input
```
**But I can't find this **forward** function in my code. Why does this happen? Please help me!**

**Comment by suo** [comment]
@kevin-steiner if you try again with the new [recursive scripting API](https://pytorch.org/blog/pytorch-1.2-and-domain-api-release/#torchscript-improvements) we released in 1.2, you will get much better error messages with source range information. We should support custom LSTMs in TorchScript so it's not immediately clear what the issue is as is.

Going to close this out as there has not been much activity and the issue is not reproducible. Feel free to reopen if you want us to take another look!
